src/Backend/WASMWrapper.js

```diff
@@ -1,27 +1,88 @@
-const __wasm = __init_wasm()
-
-//#region TODO:
-function __init_wasm() { }
-//#endregion TODO
-
-//#region MOCK
-//#region GET
-export function getTrace(particle, propagationMedium) {
-    if (__wasm) console.log("wasm");
-    return {
-        x: [10, 20, 30, 40, 50, 60, 70, 80, 90, 100],
-        y: Array.from(new Array(10),_=>Math.random()*100)
+import Module from './weblibdedx.js'
+
+export default class WASMWrapper {
+    #_wasm;
+    #programsSize = 20
+    #ionsSize = 20
+    #materialsSize = 200
+
+    async wasm() {
+        if (!this.#_wasm) this.#_wasm = await Module()
+        return this.#_wasm
+    }
+
+    async getNames(values, func) {
+        const wasm = await this.wasm()
+        const getName = wasm.cwrap(func, 'number', ['number'])
+        return values.map(val => {
+            return {
+                code: val,
+                name: wasm.UTF8ToString(getName(val))
+            }
+        })
+    }
```

Comment:
can those 2 functions be private?

Reply:
I'd love to but babel, which is a module loader react uses, doesn't support private class member functions - they are considered an experimental feature and adding the support is much more hustle than it's worth. In the end it's not a module that's going to be exported to public